lib/rbac/access.rb

```diff
@@ -1,123 +1,83 @@
 module RBAC
   class Access
     include RBAC::Permissions
-    attr_reader :acls, :approver_acls, :owner_acls
 
-    def initialize(resource, verb)
-      @resource      = resource
-      @verb          = verb
-      @app_name      = ENV["APP_NAME"] || "approval"
-      @admin         = false
-      @approver      = false
-      @owner         = false
-      @acls          = []
-      @approver_acls = []
-      @owner_acls    = []
-    end
-
-    def process
-      RBAC::Service.call(RBACApiClient::AccessApi) do |api|
-        Rails.logger.info("Fetch access list for #{@app_name}")
-        full_acls = RBAC::Service.paginate(api, :get_principal_access, {}, @app_name)
-
-        regexp = Regexp.new(":(#{@resource}|\\*):(#{@verb}|\\*)")
-        approver_regexp = Regexp.new(":(workflows|\\*):(approve|\\*)")
-
-        full_acls.each do |item|
-          Rails.logger.debug("Found ACL: #{item}")
-          if regexp.match(item.permission)
-            @acls << item
-            unless @admin
-              item.resource_definitions.any? do |rd|
-                @admin = true if rd.attribute_filter.key == 'id' &&
-                                 rd.attribute_filter.operation == 'equal' &&
-                                 rd.attribute_filter.value == '*'
-              end
-            end
-          end
-
-          @approver_acls << item if approver_regexp.match(item.permission)
-        end
-
-        @approver = true if approver_acls.any?
-
-        @owner_acls = requester_acls.select do |item|
-          regexp.match(item.permission)
-        end
+    ADMIN_ROLE = 'Approval Administrator'.freeze
```

Comment:
@hsong-rh Since this is shared code should we put anything specific to Approval in here. At some point this is going to be in the shared common gem. The Admin would be different for Catalog/Approval/Topology and other service.

Reply:
Will change when the real sharing starts.